Kernel/Process.cpp

```diff
@@ -2384,6 +2510,24 @@ pid_t Process::sys$getppid()
     return m_ppid;
 }
 
+int Process::sys$getresuid(uid_t* ruid, uid_t* euid, uid_t* suid)
+{
+    REQUIRE_PROMISE(stdio);
+    *ruid = m_uid;
+    *euid = m_euid;
```

Comment:
Here and below: you can't just dereference user pointers like that 😀

You have to `validate_write_typed()`, `return EFAULT`, and `copy_to_user`. Otherwise this is an arbitrary kernel write primitive 🙂